test/end2end_test.go

```diff
@@ -1037,7 +1051,10 @@ func (s) TestDetailedConnectionCloseErrorPropagatesToRpcError(t *testing.T) {
 	// the RPC has been started on it.
 	<-rpcStartedOnServer
 	ss.S.Stop()
-	if _, err := stream.Recv(); err == nil || (!strings.Contains(err.Error(), possibleConnResetMsg) && !strings.Contains(err.Error(), possibleEOFMsg)) {
+	// The precise behavior of this test is subject to raceyness around the timing
+	// of when TCP packets are sent from client to server, and when we tell the
+	// server to stop, so we need to account for both possible error messages.
+	if _, err := stream.Recv(); err == nil || !isConnClosedErr(err) {
```

Comment:
I would still combine these.

`isConnClosedErr(nil)` should be false. And actually it is now.  So I think just remove the `err == nil ||` here?

Reply:
sure that makes sense